### PR TITLE
MCR-2135 add support for order and classification when adding derivate

### PR DESCRIPTION
**Pull request by yagee-de**
[Link to jira](https://mycore.atlassian.net/browse/MCR-2135).

### CODE REVIEW

**Comment by adlerre** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- mycore-restapi/src/main/java/org/mycore/restapi/converter/MCRCategoryIDMessageBodyReader.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5510189&bid=18059728)

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on b2c47acd1c0defa634a5be121eacbd1730a4b4af! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/5518-pull_request-963

File: mycore-restapi/src/main/java/org/mycore/restapi/v2/MCRRestDerivates.java [hunk 1]
```diff
@@ -394,29 +457,60 @@ boolean isDisplayEnabled() {
...
+        }
+
         String getMainDoc() {
             return mainDoc;
         }
 
-        @FormDataParam("main-doc")
+        @FormParam("main-doc")
         @JsonProperty("main-doc")
         public void setMainDoc(String mainDoc) {
```
Comment (rsteph-de) [inline mycore-restapi/src/main/java/org/mycore/restapi/v2/MCRRestDerivates.java:473]: param name should be "maindoc" as in derivate xml or xml of MCRMetaEnrichedLinkID

File: mycore-restapi/src/main/java/org/mycore/restapi/v2/MCRRestDerivates.java [hunk 2]
```diff
@@ -394,29 +457,60 @@ boolean isDisplayEnabled() {
...
+        }
+
+        @JsonProperty
+        public void setOrder(int order) {
+            this.order = order;
+        }
+
+        @FormParam("order")
+        @DefaultValue("1")
+        public void setOrder(String order) {
```
Comment (rsteph-de) [inline mycore-restapi/src/main/java/org/mycore/restapi/v2/MCRRestDerivates.java]: @FormParam may also work with setOrder(int order)
https://mkyong.com/webservices/jax-rs/jax-rs-formparam-example/
no parse-from-string necessary

File: mycore-restapi/src/main/java/org/mycore/restapi/v2/MCRRestDerivates.java [hunk 3]
```diff
@@ -394,29 +457,60 @@ boolean isDisplayEnabled() {
...
         }
 
-        @FormDataParam("display-enabled")
         @JsonProperty("display-enabled")
         public void setDisplayEnabled(boolean displayEnabled) {
             this.displayEnabled = displayEnabled;
         }
 
+        @FormParam("display-enabled")
+        @DefaultValue("true")
```
Comment (rsteph-de) [inline mycore-restapi/src/main/java/org/mycore/restapi/v2/MCRRestDerivates.java]: the corresponding attribute in derivate xml is called "display" only - why is another name used here?

**Review by rsteph-de — CHANGES_REQUESTED** [review]
please have a look at the names of the parameters - they should match the MyCoRe XML as close as possible.

File: mycore-restapi/src/main/java/org/mycore/restapi/v2/MCRRestDerivates.java (hunk 3 above)
Comment (yagee-de) [reply mycore-restapi/src/main/java/org/mycore/restapi/v2/MCRRestDerivates.java]: it's derived from the method name "setDisplayEnabled", but I can rename it to display

**Review by rsteph-de — APPROVED** [review]
looks good for me now

**Review by kkrebs — APPROVED** [review]

**Pull request merged**